File: backend/components/metiers/threed/security/validators/rgpd/rgpd_validator.js
// rgpd_validator.js – Validation conformité RGPD

function validateRGPD(data) {
  if ('personal_data' in data && !data.consent) throw new Error('Non conforme RGPD : consentement manquant');
  return true;
}

module.exports = { validateRGPD };

File: backend/components/metiers/threed/security/validators/rgpd/rgpd_validator.py
# rgpd_validator.py – Validation conformité RGPD


def validate_rgpd(data):
    # Exemple de validation RGPD
    if "personal_data" in data and not data.get("consent", False):
        raise ValueError("Non conforme RGPD : consentement manquant")
    return True
